lib/internal/http2/core.js

```diff
@@ -1338,6 +1371,40 @@ class ServerHttp2Session extends Http2Session {
 
     this[kHandle].altsvc(stream, origin || '', alt);
   }
+
+  // Submits an origin frame to be sent.
+  origin(...origins) {
+    if (this.destroyed)
+      throw new ERR_HTTP2_INVALID_SESSION();
+
+    if (origins.length === 0)
+      return;
+
+    let arr = '';
+    let len = 0;
+    const count = origins.length;
+    for (var i = 0; i < count; i++) {
+      let origin = origins[i];
+      if (typeof origin === 'string') {
+        origin = (new URL(origin)).origin;
+      } else if (origin != null && typeof origin === 'object') {
```

Comment:
Just a quick question.
How can you be sure here that the object will have the origin attribute?

And an error is thrown below even though this assignment goes through. I just came here by chance and was wondering about it :)

Reply:
If the object does not have an `origin` property, then the `origin` variable will be assigned as `undefined`, which will be caught by the following `typeof origin !== 'string'` check. So, we aren't sure if the object has one, we optimistically assign then check after.

Reply:
Thanks :+1: